### PR TITLE
remove duplicate static listener conflict checks

### PR DESCRIPTION
**Pull request by ramaraochavali**
**Please provide a description of this PR:**

### CODE REVIEW

File: pilot/pkg/model/sidecar_test.go [hunk 1]
```diff
@@ -2503,6 +2554,9 @@ func TestCreateSidecarScope(t *testing.T) {
 	}
 
 	for _, tt := range tests {
+		if tt.name != "multi-port-merge-in-same-namespace" {
```
[the remark this answers is not in the document]
Comment (ramaraochavali) [reply pilot/pkg/model/sidecar_test.go:2557]: Sorry. Missed to revert this change

File: pilot/pkg/networking/core/listener_test.go [hunk 2]
```diff
@@ -1984,7 +1984,7 @@ func testInboundListenerConfigWithSidecarConflictPort(t *testing.T, proxy *model
 						Name:     "uds",
 					},
 					CaptureMode:     2, // None
-					Bind:            "1.1.1.1",
+					Bind:            "0.0.0.0",
```
Comment (ramaraochavali) [inline pilot/pkg/networking/core/listener_test.go:1987]: This seems incorrect test. With bind "1.1.1.1" should not cause conflict?

File: pilot/pkg/model/context.go [hunk 3]
```diff
@@ -811,23 +811,18 @@ func (node *Proxy) IsUnprivileged() bool {
 	return unprivileged
 }
 
-// CanBindToPort returns true if the proxy can bind to a given port.
-func (node *Proxy) CanBindToPort(bindTo bool, port uint32) bool {
+// CanBindToPrivilegedPort returns true if the proxy can bind to a given port.
+func (node *Proxy) CanBindToPrivilegedPort(bindTo bool, port uint32) bool {
```
Comment (hzxuzhonghu) [inline pilot/pkg/model/context.go:815]: I believe the older name is more correct

File: pilot/pkg/model/context.go [hunk 4]
```diff
@@ -811,23 +811,18 @@ func (node *Proxy) IsUnprivileged() bool {
...
 	if bindTo {
-		if IsPrivilegedPort(port) && node.IsUnprivileged() {
-			return false
-		}
-		if node.Metadata != nil &&
-			(node.Metadata.EnvoyPrometheusPort == int(port) || node.Metadata.EnvoyStatusPort == int(port)) {
```
Comment (hzxuzhonghu) [inline pilot/pkg/model/context.go:821]: why remove the static check

**Review by hzxuzhonghu — COMMENTED** [review]
I would suggest merge `conflictWithReservedListener` into `CanBindPort`, so that we can only check once

**Comment by ramaraochavali** [comment]
> I would suggest merge conflictWithReservedListener into CanBindPort, so that we can only check once

Done. PTAL

**Review by hzxuzhonghu — APPROVED** [review]
/LGTM

**Pull request merged**